### DISCUSSION TITLE
HLS m3u8 file contain some outrageous file segments

### DISCUSSION THREAD

**Discussion by lexy10** [ant-media/Ant-Media-Server#6645] [Q&A]
I am confused on what is going on, when streaming hls and uploading to s3 or on local server, after streaming for 10mins, the m3u8 #EXTINF: file contains some large number. Example below

#EXTINF:2.800000,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000036.ts
#EXTINF:1.433333,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000037.ts
#EXTINF:1.333333,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000038.ts
#EXTINF:2.766667,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000039.ts
#EXTINF:1.400000,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000040.ts
#EXTINF:2.433333,
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000041.ts
**#EXTINF:89483.400000,**
livestream_b2c9a158-0c91-4860-9cb0-ed248997689d_1080p2500kbps000000042.ts
#EXTINF:2.400000,

the average is 2.4, but then we have #EXTINF:89483.400000,

And it happens in like 3 places depending on how long the stream is, how can this be solved please? I am using adaptive rate too.

**Comment by lastpeony** [comment]
Can you share your publish settings, enviroment and ams version?
Specifically, your encoder, FPS, GOP, and bitrate.
Which software are you using to publish?
Are you publishing from an IP camera?
What protocol are you using for publishing (RTMP, RTSP, WebRTC, SRT)?

**Reply by lexy10** [reply]
Environment: Docker
AMS Version: 2.10.0 Enterprise
Encoder: H.264
FPS: 30
GOP: No idea
Bitrates: 800hbps, 1000kbps, 1800kbps, 2000kbps, 2500kbps (I am using the Adaptive Streaming)

I publish from Android and IOS SDK through WebRTC

**Reply by lastpeony** [reply]
There are few similar issues to this which we are currently investigating.
https://github.com/ant-media/Ant-Media-Server/issues/6506

Converted this discussion to an issue.
https://github.com/ant-media/Ant-Media-Server/issues/6652

Have you ever faced with this issue without adaptive streaming?

**Reply by lexy10** [reply]
I have not really tested without adaptive streaming because we really need the adaptive streaming

**Reply by lastpeony** [reply]
I will try to reproduce this on my end. thanks

**Reply by lexy10** [reply]
Please is there any heads up here?

**Reply by lastpeony** [reply]
Can you please upgrade to 2.11.3 ? In an another case we managed to fix it by upgrading. There is an improvement regarding to webrtc to hls in latest version and it should fix this problem also.
@lexy10

**Reply by lexy10** [reply]
Thank you for the update, using docker to build the ant-media, please how can I upgrade to the latest version without loosing the settings made already...

**Reply by lastpeony** [reply]
https://antmedia.io/docs/guides/installing-on-linux/upgrading-ant-media-server/
You can use upgrade.sh script

**Reply by lexy10** [reply]
Alright got it, I'll upgrade and monitor it for some period of time and definitely get back to you. Thanks once again.

**Reply by lexy10** [reply]
Trying to upgrade to the latest version, I get this error after accepting to replace all files

mv: cannot move '/usr/local/antmedia' to '/usr/local/antmedia-backup-2024-09-30_17-04-35': Device or resource busy
There is a problem in installing the ant media server. Please send the log of this console to support@antmedia.io

What could be the issue?

**Reply by filip-ams** [reply]
Hi @lexy10,

This message usually indicates permission issues, did you run the command with sudo privileges? Also to keep your settings you can follow the Upgrading with the Installation Script in our [documentation](https://antmedia.io/docs/guides/installing-on-linux/upgrading-ant-media-server/#upgrading-with-the-upgradesh-script). I hope this helps.

**Reply by lexy10** [reply]
I am running the upgrade inside of a docker container so there's no sudo needed. But still shows the error...
I will try to make use of the installation script then.

**Reply by lexy10** [reply]
Hello @filip-ams 

I have tried the installation script too, I got the same error. I am so confused on what to do now. I think the issue is running inside docker since there's no sudo inside docker container.

**Reply by filip-ams** [reply]
Hi @lexy10,

Could you try to run your docker container with --privileged flag ? The command should look something like this:

docker run --privileged --restart=always -d --name antmedia --network=host -it antmediaserver

You can read more about the --privileged flag [here](https://docs.docker.com/engine/containers/run/#runtime-privilege-and-linux-capabilities).

**Reply by lexy10** [reply]
Hello @filip-ams 

Thank you for the response, I have tried everything possible including the --privileged flag, i still get the same thing. This is my command below

docker run --privileged --restart=always -d --name antmedia --mount source=antmedia_volume,target=/usr/local/antmedia/ --network=host -it antmediaserver

And I still get this response:

mv: cannot move '/usr/local/antmedia' to '/usr/local/antmedia-backup-2024-10-02_10-47-16': Device or resource busy
There is a problem in installing the ant media server. Please send the log of this console to support@antmedia.io
